### PR TITLE
Require Node.js 8, add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	readonly encoding?: EncodingType;
+
+	/**
+	 * Values: `md5` `sha1` `sha256` `sha512` *([Platform dependent](https://nodejs.org/api/crypto.html#crypto_crypto_createhash_algorithm))*
+	 *
+	 * *The `md5` algorithm is good for [file revving](https://github.com/sindresorhus/rev-hash), but you should never use `md5` or `sha1` for anything sensitive. [They're insecure.](http://googleonlinesecurity.blogspot.no/2014/09/gradually-sunsetting-sha-1.html)*
+	 *
+	 * @default 'sha512'
+	 */
+	readonly algorithm?: string;
```
Comment (sindresorhus) [inline index.d.ts]: Can you list the encodings here? It says "platform dependent", but I'm pretty sure these algorithms are available everywhere.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	 *
+	 * @returns The created hash transform stream.
+	 */
+	stream(options?: Options<HashaEncoding>): Hash;
+
+	/**
+	 * Calculate the hash for a stream.
+	 *
+	 * @param stream - A stream you want to hash.
+	 * @returns A `Promise` for a hash.
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	 * @returns A hash.
```

File: index.d.ts (hunk 1 above)
Comment (BendingBender) [reply index.d.ts]: But should I still allow plain strings?

If so, I don't think that it makes a difference (IDE-wise) to list algorithms when allowing plain strings. There will be no auto-completion for the listed algorithms.

I could, of course do it anyway.

Comment (sindresorhus) [reply index.d.ts]: Yes, it should still allow plain string. There's a workaround to get this working, see: https://github.com/sindresorhus/boxen/pull/39#discussion_r252933967

Comment (BendingBender) [reply index.d.ts]: Wow!

**Comment by sindresorhus** [comment]
Would be great if you could comment your support in https://github.com/Microsoft/TypeScript/issues/29729 The more comments they get, the more likely they are to realize they need to fix it.

**Pull request merged**